### ISSUE TITLE
💅 <2.0 beta> `useExhaustiveSwitchCases` doesn't work on enums

### ISSUE THREAD

**Issue by jpike88** [biomejs/biome#6111]
### Environment information

```bash
https://typescript-eslint.io/rules/switch-exhaustiveness-check/

This rule says it works on both unions and enums. Now biome is becoming type aware, this rule needs to be updated to handle enums
```

### Rule name

2.0 beta 5

### Playground link

https://next.biomejs.dev/playground/?lintRules=useExhaustiveSwitchCases&code=CgBlAG4AdQBtACAARABhAHkAIAB7AAoAIAAgAE0AbwBuAGQAYQB5ACAAPQAgACcATQBvAG4AZABhAHkAJwAsAAoAIAAgAFQAdQBlAHMAZABhAHkAIAA9ACAAJwBUAHUAZQBzAGQAYQB5ACcALAAKACAAIABXAGUAZABuAGUAcwBkAGEAeQAgAD0AIAAnAFcAZQBkAG4AZQBzAGQAYQB5ACcALAAKACAAIABUAGgAdQByAHMAZABhAHkAIAA9ACAAJwBUAGgAdQByAHMAZABhAHkAJwAsAAoAIAAgAEYAcgBpAGQAYQB5ACAAPQAgACcARgByAGkAZABhAHkAJwAsAAoAIAAgAFMAYQB0AHUAcgBkAGEAeQAgAD0AIAAnAFMAYQB0AHUAcgBkAGEAeQAnACwACgAgACAAUwB1AG4AZABhAHkAIAA9ACAAJwBTAHUAbgBkAGEAeQAnACwACgB9AAoACgBkAGUAYwBsAGEAcgBlACAAYwBvAG4AcwB0ACAAZABhAHkAOgAgAEQAYQB5ADsACgBsAGUAdAAgAHIAZQBzAHUAbAB0ACAAPQAgADAAOwAKAAoAcwB3AGkAdABjAGgAIAAoAGQAYQB5ACkAIAB7AAoAIAAgAGMAYQBzAGUAIAAnAE0AbwBuAGQAYQB5ACcAOgAKACAAIAAgACAAcgBlAHMAdQBsAHQAIAA9ACAAMQA7AAoAIAAgACAAIABiAHIAZQBhAGsAOwAKAH0ACgAKAA%3D%3D

### Expected result

it should look at enums that are used for cases and ensure all scenarios get covered.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by arendjr** [comment]
Correct, we don't have parity on that front yet. Enums would definitely be a good thing for us to implement still!

**Comment by wanghaoPolar** [comment]
hi @arendjr , I'm very interested in this task, can you help to assign it to me? Thanks!

**Comment by siketyan** [comment]
@wanghaoPolar All yours! Thank you for picking up this 🙏

**Comment by coupez** [comment]
Any updates on this? Still not working on enums in 2.2.6.
